weaver/store/mongodb.py

```diff
@@ -582,6 +590,25 @@ def find_jobs(self,
         if service is not None:
             search_filters["service"] = service
 
+        if datetime_interval is not None:
+            datetime_interval = datetime_interval_parser(datetime_interval)
+            query = {}
+
+            try:
+                if datetime_interval.get("after", False):
+                    query["$gte"] = datetime_interval["after"]
+
+                if datetime_interval.get("before", False):
+                    query["$lte"] = datetime_interval["before"]
+
+                if datetime_interval.get("match", False):
+                    query = datetime_interval["match"]
+
+            except Exception as ex:
+                raise JobRegistrationError("Error occurred during datetime job filtering: [{}]".format(repr(ex)))
+
+            search_filters["created"] = query
```

Comment:
What about invalid cases where start-datetime >= end-datetime ? 
Should be tested and raised if invalid.

Reply:
Since we used `$lte` and `$gte`, it should return the same result as a `match ` query, should we remove that behavior and only keep `$lt` and `$gt`?